scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java

```diff
@@ -1192,6 +1192,28 @@ public void removeThirdPartyCredential(String sessionId, String key) throws Rest
         }
     }
 
+    private Map<String, Object> executeGraphQLQuery(String sessionId, String query, String variables)
+            throws ServiceException, RestServerException {
+
+        if (sessionId == null || query == null)
+            return null;
+
+        ResteasyClient client = new ResteasyClientBuilder().asyncExecutor(threadPool)
+                                                           .httpEngine(new ApacheHttpClient4Engine(httpClient))
+                                                           .build();
+        ResteasyWebTarget target = client.target(SchedulerConfig.get().getSchedulingApiUrl());
+
+        GraphQLClient graphQLClientProxy = target.proxy(GraphQLClient.class);
```

Comment:
L1201 to 1206 would be better to be moved to Scheduling-Api-Client module as well.